asmrun/natdynlink.c

```diff
@@ -35,6 +35,14 @@ CAMLexport void (*caml_natdynlink_hook)(void* handle, char* unit) = NULL;
 
 #include <stdio.h>
 #include <string.h>
+#include <limits.h>
+
+#define Handle_val(v) (*((void **) (v)))
```

Comment:
I think `Field` is the wrong thing to use on `Abstract_tag` blocks, since they're not composed of fields (it'll work fine on trunk, but break on multicore).

For this purpose, multicore has `Data_abstract_val` in `mlvalues.h`. I've added it to the pull rq (its usage is the same as `Data_custom_val`).